### PR TITLE
[ACR] Adds support for Azure DevOps URLs in build/run

### PR DESCRIPTION
**Pull request by ehotinger**
Allows users to call `az acr build` and `az acr run` against Azure DevOps URLs.

/cc @djyou @northtyphoon 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_archive_utils.py [hunk 1]
```diff
@@ -14,6 +14,9 @@
 from msrestazure.azure_exceptions import CloudError
 from azure.storage.blob import BlockBlobService
 from ._azure_utils import get_blob_info
+from ._constants import (
+    TASK_VALID_VSTS_URLS,
```
Comment (djyou) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_archive_utils.py]: NIT: can be in one line.

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_constants.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 BUILD_STEP_RESOURCE_TYPE = BUILD_TASK_RESOURCE_TYPE + '/steps'
 
 TASK_RESOURCE_TYPE = REGISTRY_RESOURCE_TYPE + '/tasks'
+TASK_VALID_VSTS_URLS = ["visualstudio.com", "dev.azure.com"]
```
Comment (djyou) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_constants.py]: NIT: single quote

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_archive_utils.py [hunk 3]
```diff
@@ -208,8 +209,15 @@ def check_remote_source_code(source_location):
     # http
     if lower_source_location.startswith("https://") or lower_source_location.startswith("http://") \
        or lower_source_location.startswith("github.com/"):
-        if re.search(r"\.git(?:#.+)?$", lower_source_location) or "visualstudio.com" in lower_source_location:
-            # git url must contain ".git" or be from VSTS
+        isVSTS = False
+        for vstsURL in TASK_VALID_VSTS_URLS:
+            if vstsURL in lower_source_location:
+                isVSTS = True
+                break
```
Comment (williexu) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_archive_utils.py]: `isVSTS = any(url in lower_source_location for url in task_valid_urls)`

Comment (ehotinger) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_archive_utils.py]: Thanks, good one liner.

**Comment by tjprescott** [comment]
@williexu please take a second look.

**Pull request merged**